tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -65,6 +73,8 @@ class DistributedTransformerTest
       optimization_guard_;
   preseg_passes::OptimizationPassGuard<preseg_passes::AllocationDomainPass>
       allocation_order_guard_;
+  preseg_passes::OptimizationPassGuard<preseg_passes::MarkAliasesPreparePass>
```

Comment:
FYI @wujingyue - https://github.com/NVIDIA/Fuser/pull/2639 was causing issues with transpose. It might be as simple as propagating shardings after the alias pass. I will get a small reproducible test.

Reply:
> The fix might be as simple as propagating shardings after the alias pass. 

I don't see how that's possible with MultiDeviceExecutor. Yes with FusionExecutorCache. Nonetheless, I'll wait for your repro to understand the problem better.